server/src/main/java/org/apache/uniffle/server/storage/HadoopStorageManager.java

```diff
@@ -149,7 +193,19 @@ public void removeResources(PurgeEvent event) {
                   storage.getStoragePath()));
         }
       }
-      deleteHandler.delete(deletePaths.toArray(new String[0]), appId, event.getUser());
+      if (isQuick) {
+        AsynchronousDeleteEvent asynchronousDeleteEvent =
+            new AsynchronousDeleteEvent(
+                appId, event.getUser(), storage.getConf(), event.getShuffleIds(), deletePaths);
+        deleteHandler.quickDelete(asynchronousDeleteEvent);
+        boolean isSucess = quickNeedDeletePaths.offer(asynchronousDeleteEvent);
+        if (!isSucess) {
+          LOG.warn(
```

Comment:
This is abnormal that will make the data leaked. For this case, the metrics should be added for better observability